backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/SchedulingManager.java

```diff
@@ -578,29 +605,42 @@ private void addPendingResources(VM vm, VDS host, Map<Integer, NumaNodeMemoryCon
 
         Map<Guid, Map<Integer, NumaNodeMemoryConsumption>> result = new HashMap<>();
         for (VM vm : filteredVms) {
-            if (vm.getvNumaNodeList().isEmpty()) {
-                continue;
-            }
+            vmNumaRequirements(vm, nodeAssignment, result);
+        }
 
-            Map<Integer, NumaNodeMemoryConsumption> hostNumaMemRequirements = new HashMap<>();
-            Optional<Integer> hugePageSize = HugePageUtils.getHugePageSize(vm.getStaticData());
+        return result;
+    }
+
+    private void vmNumaRequirements(VM vm, Optional<Map<Guid, Integer>> nodeAssignment,
+                                    Map<Guid, Map<Integer, NumaNodeMemoryConsumption>> result) {
```

Comment:
so why not to set this method to return the consumption?

Reply:
I decided to have it like that because the call in line 608. There (normal run), we can have a group of VMs. This function deal with 1 VM.

Reply:
so we can collect the results, no?

Reply:
and formatter

Reply:
only formatter done